meson.build

```diff
@@ -0,0 +1,53 @@
+project(
+  'pgroonga',
+  'c',
+  license: 'PostgreSQL',
+  version: '4.0.2')
+
+required_groonga_version = '14.0.0'
+
+groonga = dependency('groonga', version: '>= ' + required_groonga_version)
+cc = meson.get_compiler('c')
+libm = cc.find_library('m')
+
+pg_config_opt = get_option('pg_config')
+pg_config = find_program(pg_config_opt, 'pg_config')
```

Comment:
What does happen when `-Dpg_config` isn't specified?

Reply:
If `-Dpg_config` isn’t specified, Meson will use whatever `pg_config_option` we’ve set, and if that lookup fails it will fall back to searching for `pg_config` in our PATH.
https://mesonbuild.com/Reference-manual_functions.html#find_program

Reply:
> If `-Dpg_config` isn’t specified, Meson will use whatever `pg_config_option` we’ve set,

If we don't specify `-Dpg_config`, does "whatever `pg_config_option` we’ve set" exist?
It seems that the default value (an empty string) is used for the case. Is it expected?

Reply:
Yes, I expected. You’re right that if the option isn’t specified it leads to unnecessary lookups, so I’ll add a conditional check to skip the search when no `-Dpg_config` is provided.